### PR TITLE
Refactor SVar module

### PR DESCRIPTION
**Pull request by harendra-kumar**
Review can start with the module structure changes in streamly.cabal. This is a major refactor of the SVar module and some other high level modules associated with concurrency.

* Break it up into smaller modules with clearly separated functionality
* Keep minimal stuff in the lower level module SVar/Type, and push the
  rest in a higher level module to reduce dependencies of lower level
  modules. This will help in separating the lower modules in streamly-core
  package.

### CODE REVIEW

**Comment by harendra-kumar** [comment]
I compared the benchmarks before and after the refactor, there is no noticeable change.

File: src/Streamly/Internal/Data/SVar/Worker.hs [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+    (
+    -- * Adjusting Limits
+      decrementYieldLimit
+    , incrementYieldLimit
+    , decrementBufferLimit
+    , incrementBufferLimit
+    , resetBufferLimit
+
+    -- * Rate Control
+    , Work (..)
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/SVar/Worker.hs:20]: Should `WorkInfo` also be in this module?

File: src/Streamly/Internal/Data/SVar/Worker.hs [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+    tid <- myThreadId
+    void $ sendToProducer sv (ChildStop tid Nothing)
+
+-------------------------------------------------------------------------------
+-- Send exceptions
+-------------------------------------------------------------------------------
+
+{-# NOINLINE handleFoldException #-}
+handleFoldException :: SVar t m a -> SomeException -> IO ()
+handleFoldException sv e = do
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/SVar/Worker.hs:536]: The consumer fold?

File: src/Streamly.hs [hunk 3]
```diff
@@ -210,7 +210,8 @@ module Streamly {-# DEPRECATED "Please use \"Streamly.Prelude\" instead." #-}
 where
 
 import Data.Semigroup (Semigroup(..))
-import Streamly.Internal.Data.SVar (MonadAsync, Rate(..))
+import Streamly.Internal.Control.Concurrent (MonadAsync)
+import Streamly.Internal.Data.SVar (Rate(..))
```
Comment (adithyaov) [inline src/Streamly.hs:214]: We'll have to maintain deprecated modules? Do we remove them after a few releases after deprecating them?

**Review by adithyaov — APPROVED** [review]

**Comment by adithyaov** [comment]
Looks good

File: src/Streamly.hs (hunk 3 above)
Comment (harendra-kumar) [reply src/Streamly.hs:214]: We will remove after a while.

File: src/Streamly/Internal/Data/SVar/Worker.hs (hunk 2 above)
Comment (harendra-kumar) [reply src/Streamly/Internal/Data/SVar/Worker.hs:536]: Yes.

File: src/Streamly/Internal/Data/SVar/Worker.hs (hunk 1 above)
Comment (harendra-kumar) [reply src/Streamly/Internal/Data/SVar/Worker.hs:20]: No it is used in the SVar type definition so has to be in SVar.Type.

**Pull request merged**